src/components/app.tsx

```diff
@@ -49,6 +58,23 @@ class App extends React.Component<Props & {match: any, location: any}> {
     const parameter = this.props.match.params;
     this.setSpecInUrl(parameter);
   }
+  /**
+   * Return Promise with updated link via vega-loader
+   */
+  public updateLink(url: string): any {
+    const editorURL = 'https://vega.github.io';
+    const loader = vega.loader();
+    const options = {
+      baseURL: this.props.baseURL
```

Comment:
Instead of forcing the baseURL, try to check for every request whether it makes sense to change the baseURL. First, check whether there is data at the `evt.origin` that you grabbed above. If not, use the editor as the base.